File: frontend/src/types/feedback.ts
/**
 * 피드백 종류를 나타내는 타입
 * @typedef {"bug" | "feature" | "improvement" | "general"} FeedbackType
 */
export type FeedbackType = "bug" | "feature" | "improvement" | "general";

/**
 * 피드백 전송 요청 데이터 인터페이스
 * @interface FeedbackRequest
 * @property {string} email - 사용자 이메일 주소
 * @property {FeedbackType} type - 피드백 종류
 * @property {string} message - 피드백 내용
 * @property {string} [url] - 현재 페이지 URL (선택사항)
 * @property {string} [userAgent] - 브라우저 정보 (선택사항)
 */
export interface FeedbackRequest {
  email: string;
  type: FeedbackType;
  message: string;
  url?: string;
  userAgent?: string;
}

/**
 * 피드백 전송 응답 데이터 인터페이스
 * @interface FeedbackResponse
 * @property {boolean} success - 전송 성공 여부
 * @property {string} message - 응답 메시지
 */
export interface FeedbackResponse {
  success: boolean;
  message: string;
}

/**
 * 피드백 타입 선택 옵션 인터페이스
 * @interface FeedbackTypeOption
 * @property {FeedbackType} value - 피드백 타입 값
 * @property {string} label - 화면에 표시될 라벨
 * @property {string} icon - 아이콘 이모지
 */
export interface FeedbackTypeOption {
  value: FeedbackType;
  label: string;
  icon: string;
} 

File: backend/feedback/model/feedback.go
// Package model은 피드백 관련 데이터 구조를 정의합니다.
package model

import "time"

// FeedbackRequest는 클라이언트로부터 받는 피드백 요청 구조입니다.
type FeedbackRequest struct {
	Type    string `json:"type" binding:"required,oneof=bug feature general"` // 피드백 타입
	Message string `json:"message" binding:"required,min=10,max=1000"`        // 피드백 내용
	Email   string `json:"email,omitempty" binding:"omitempty,email,max=100"` // 선택적 연락처
}

// FeedbackResponse는 클라이언트에게 보내는 응답 구조입니다.
type FeedbackResponse struct {
	Success bool   `json:"success"`
	Message string `json:"message"`
	ID      string `json:"id,omitempty"` // 추적용 ID
}

// FeedbackLog는 로깅용 구조체입니다.
type FeedbackLog struct {
	ID        string    `json:"id"`
	Type      string    `json:"type"`
	ClientIP  string    `json:"client_ip"`
	Success   bool      `json:"success"`
	Error     string    `json:"error,omitempty"`
	Timestamp time.Time `json:"timestamp"`
}
